src/Writers/HTMLWriter.jl

```diff
@@ -4,24 +4,471 @@ Provides the [`render`](@ref) methods to write the documentation as HTML files
 """
 module HTMLWriter
 
+using Compat
+
 import ...Documenter:
     Anchors,
     Builder,
     Documents,
     Expanders,
     Formats,
     Documenter,
-    Utilities
+    Utilities,
+    Writers
 
 import ..Writers: Writer, render
 import ...Utilities.DOM: DOM, Tag, @tags
+using ...Utilities.MDFlatten
+
+const requirejs_cdn = "https://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.min.js"
+
+type HTMLContext
+    doc :: Documents.Document
+    logo :: Compat.String
+    scripts :: Vector{Compat.String}
+    documenter_js :: Compat.String
+    documenter_css :: Compat.String
+    search_index :: IOBuffer
+    search_index_js :: Compat.String
+end
+HTMLContext(doc) = HTMLContext(doc, "", [], "", "", IOBuffer(), "")
+
+"""
+Returns a page from the [`HTMLContext`](@ref).
+"""
+getpage(ctx, path) = ctx.doc.internal.pages[path]
+getpage(ctx, navnode::Documents.NavNode) = getpage(ctx, get(navnode.page))
 
-# TODO
 
 function render(::Writer{Formats.HTML}, doc::Documents.Document)
-    error("HTML rendering is unsupported.")
+    !isempty(doc.user.sitename) || error("HTML output requires `sitename`.")
+
+    ctx = HTMLContext(doc)
+    ctx.search_index_js = "search_index.js"
+
+    copy_asset("layout.css", doc)
+    copy_asset("style.css", doc)
+    ctx.documenter_css = copy_asset("documenter.css", doc)
+
+    local_assetsdir = joinpath(pwd(), "assets")
+    let src = joinpath(local_assetsdir, "logo.png")
+        if isfile(src)
+            dst = joinpath(doc.user.build, "logo.png")
+            cp(src, dst, remove_destination=true)
+            ctx.logo = "logo.png"
+        end
+    end
+
+    ctx.documenter_js = copy_asset("documenter.js", doc)
+
+    for navnode in doc.internal.navlist
+        render_page(ctx, navnode)
+    end
+
+    open(joinpath(doc.user.build, ctx.search_index_js), "w") do io
+        println(io, "var documenterSearchIndex = {\"docs\": [\n")
+        write(io, takebuf_string(ctx.search_index))
+        println(io, "]}")
+    end
```

Comment:
Would be great but I think I ran into problems with the same-origin policy when I was trying to load the search index with ajax on a local build (some of the `.html` files are in a subdirectory). Hence the need to "load" it with a `<script>` tag.